src/platform-implementation-js/dom-driver/gmail/views/gmail-thread-row-view.ts

```diff
@@ -463,52 +470,74 @@ class GmailThreadRowView {
 
           imageMod = null;
         }
-      } else {
-        if (!imageMod) {
-          imageMod = this._modifications.image.unclaimed.shift();
 
-          if (!imageMod) {
-            imageMod = {
-              iconSettings: {},
-              iconWrapper: document.createElement('div'),
+        return;
+      }
 
-              remove() {
-                this.iconWrapper.remove();
-              },
-            };
-            imageMod.iconWrapper.className =
-              'inboxsdk__thread_row_icon_wrapper';
-          }
+      if (!imageMod) {
+        imageMod = this._modifications.image.unclaimed.shift();
+
+        if (!imageMod) {
+          imageMod = {
+            iconSettings: {},
+            iconWrapper: document.createElement('div'),
 
-          this._modifications.image.claimed.push(imageMod);
+            remove() {
+              this.iconWrapper.remove();
+            },
+          };
+          imageMod.iconWrapper.className = 'inboxsdk__thread_row_icon_wrapper';
+          if (iconDescriptor.image) {
+            imageMod.iconWrapper.classList.add(s.minWidthIconWrapper);
+          }
         }
 
-        const { iconSettings, iconWrapper } = imageMod;
-        updateIcon(
-          iconSettings,
-          iconWrapper,
-          false,
-          iconDescriptor.imageClass,
-          iconDescriptor.imageUrl,
+        this._modifications.image.claimed.push(imageMod);
+      }
+
+      const { iconSettings, iconWrapper } = imageMod;
+
+      updateIcon(
+        iconSettings,
+        iconWrapper,
+        false,
+        iconDescriptor.imageClass,
+        iconDescriptor.imageUrl,
+      );
+
+      if (iconDescriptor.image) {
```

Comment:
the `unmount` should happen when imageMod gets removed. 

will this be a good option?

```
if (iconDescriptor.image) {
    const child = iconWrapper.appendChild(...);
    const dispose = iconDescriptor.image({el});

    imageMod.disposeImage = dispose;
}
```

the iconDesciptor will return as a result a callback to cleanup. This is a common patter for subscription objects or similar to react useEffect.

then where imageMod is being created, add support for the disposeImage method

```
imageMod = {
            ...
            disposeImage: null,

            remove() {
              this.disposeImage?.();
              this.iconWrapper.remove();
            },
          };
```